crates/language_models/src/provider/open_ai.rs

```diff
@@ -526,6 +526,15 @@ impl OpenAiEventMapper {
         &mut self,
         event: ResponseStreamEvent,
     ) -> Vec<Result<LanguageModelCompletionEvent, LanguageModelCompletionError>> {
+        if let Some(usage) = event.usage {
```

Comment:
This will create an issue in case of the stop event as then also we get the usage data. Now the problem is that we are just returning after usage data. Now this will break the whole event chain and the thread will be in broken state. I am not sure how to solve it tbh cause no idea how openwebui handles it.